realm/src/main/java/io/realm/Realm.java

```diff
@@ -287,9 +313,235 @@ private void removeHandler(Handler handler) {
+                            // from importing correctly the handover table view
+                            transaction.advanceRead(result.versionID);
+                        }
+
+                        ArrayList<RealmResults<?>> callbacksToNotify = new ArrayList<RealmResults<?>>(result.updatedTableViews.size());
+                        // use updated TableViews pointers for the existing async RealmResults
+                        for (Map.Entry<WeakReference<RealmResults<?>>, Long> query : result.updatedTableViews.entrySet()) {
+                            WeakReference<RealmResults<?>> weakRealmResults = query.getKey();
+                            RealmResults<?> realmResults = weakRealmResults.get();
+                            if (realmResults == null) {
+                                // don't update GC'd instance
+                                asyncRealmResults.remove(weakRealmResults);
+
+                            } else {
+                                // it's dangerous to notify the callback about new results before updating
+                                // the pointers, because the callback may use another RealmResults not updated yet
+                                // this is why we defer the notification until we're done updating all pointers
+
+                                // TODO find a way to only notify callbacks if the underlying data changed compared
+                                //      to the existing value(s) for this RealmResults (use a hashCode?)
+                                callbacksToNotify.add(realmResults);
+
+                                RealmLog.d("REALM_UPDATE_ASYNC_QUERIES realm:"+ Realm.this + " updating RealmResults " + weakRealmResults);
+                                // update the instance with the new pointer
+                                realmResults.swapTableViewPointer(query.getValue());
+                            }
+                        }
+
+                        for (RealmResults<?> query : callbacksToNotify) {
+                            query.notifyChangeListeners();
+                        }
+
+                        // notify listeners only when we advanced
+                        if (compare != 0) {
+                            sendNotifications();
+                        }
+
+                        updateAsyncQueriesTask = null;
+                    }
+                    break;
+                }
+                case REALM_COMPLETED_ASYNC_FIND_FIRST: {
+                    QueryUpdateTask.Result result = (QueryUpdateTask.Result) message.obj;
+                    Set<WeakReference<RealmObject>> updatedRowKey = result.updatedRow.keySet();
+                    if (updatedRowKey.size() > 0) {
+                        WeakReference<RealmObject> realmObjectWeakReference = updatedRowKey.iterator().next();
+                        RealmObject realmObject = realmObjectWeakReference.get();
+
+                        if (realmObject != null) {
```

Comment:
Perhaps add a comment about why it is safe to just ignore the RealmObject missing. I assume it is because the GC might have collected it before the query returns?